File: testdata/cucumber-jvm/4.8.1/src/test/java/io/cucumber/jsonschema/Stepdefs.java
package io.cucumber.jsonschema;

import cucumber.api.PendingException;
import io.cucumber.java.Before;
import io.cucumber.java.After;
import io.cucumber.java.en.Given;

public class Stepdefs {
    private static int decay = 0;

    @Before("@failing_before")
    public void tagged_failing_before_hook() {
        throw new RuntimeException("failing before hook");
    }

    @After("@failing_after")
    public void tagged_failing_after_hook() {
        throw new RuntimeException("failing after hook");
    }

    @Given("/.*pass.*/")
    public void passing_step() {
    }

    @Given("/.*pending.*/")
    public void pending_step() {
        throw new PendingException();
    }

    @Given("/.*fail.*/")
    public void failing_step() {
        throw new RuntimeException("this step failed");
    }

    @Given("/.*decaying.*/")
    public void decaying_step() {
        boolean failing = decay > 0;
        decay++;
        if (failing) throw new RuntimeException("Decayed");
    }

    @Given("/I have (\\d+) cukes in my (.*)/")
    public void cukes_in_something(int count, String something) {

    }
}

File: testdata/cucumber-js/7.3.2/stepdefs.js
const { Given, Before, After } = require('@cucumber/cucumber')

Before({ tags: '@failing_before' }, () => {
  throw new Error('failing before hook')
})

After({ tags: '@failing_after' }, () => {
  throw new Error('failing after hook')
})

Given(/pass/, () => {})

Given(/pending/, () => 'pending')

Given(/fail/, () => {
  throw new Error('this step failed')
})

let decay = 0
Given(/decaying/, () => {
  const failing = decay > 0
  decay++
  if (failing) throw new Error('Decayed')
})

Given(/skip/, () => 'skipped')

Given(/I have (\d+) cukes in my (.*)/, (_count, _something) => {})
